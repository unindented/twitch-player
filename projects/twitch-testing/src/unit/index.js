import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import React from "react";
import { render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
    ...options
  );
};

const fireResizeEvent = (el, { left = 0, top = 0, width = 0, height = 0 }) => {
  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const evt = document.createEvent("Event");
  evt.initEvent("resize", true, true);
  el.dispatchEvent(evt);
};

export * from "react-testing-library";

export { customRender as render, fireResizeEvent };
