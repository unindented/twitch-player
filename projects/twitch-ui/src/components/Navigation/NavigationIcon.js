import PropTypes from "prop-types";
import React, { memo } from "react";
import { Image } from "react-native";
import { useTheme } from "../../hooks";

const NavigationIcon = memo(
  ({ source, accessibilityLabel, testID = "navigation-icon" }) => {
    const theme = useTheme();
    const size = theme.layout.navSize;
    const style = { width: size, height: size, margin: size / 4 };

    return (
      <Image
        source={{ uri: source }}
        accessibilityLabel={accessibilityLabel}
        style={style}
        testID={testID}
      />
    );
  }
);

NavigationIcon.propTypes = {
  source: PropTypes.string.isRequired,
  accessibilityLabel: PropTypes.string.isRequired,
  testID: PropTypes.string,
};

export default NavigationIcon;
