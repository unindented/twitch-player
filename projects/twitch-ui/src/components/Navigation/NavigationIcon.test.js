import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import NavigationIcon from "./NavigationIcon";

describe("NavigationIcon", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <NavigationIcon source="some-icon.png" accessibilityLabel="Some icon" />
    );
  });

  it("renders an icon", () => {
    expect(instance.getByTestId("navigation-icon")).toMatchSnapshot();
  });
});
