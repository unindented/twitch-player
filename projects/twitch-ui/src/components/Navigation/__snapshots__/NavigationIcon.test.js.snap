// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationIcon renders an icon 1`] = `
<div
  aria-label="Some icon"
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-overflowX-11yh6sk rn-overflowY-buy8e9 rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  data-testid="navigation-icon"
  style="height: 32px; margin: 8px 8px 8px 8px; width: 32px;"
>
  <div
    class="rn-alignItems-1oszu61 rn-backgroundColor-1niwhzg rn-backgroundPosition-vvn4in rn-backgroundRepeat-u6sd8q rn-backgroundSize-4gszlv rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-bottom-1p0dtai rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-1pi2tsx rn-left-1d2f490 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-width-13qz1uu rn-zIndex-1wyyakw"
    style="background-image: url(some-icon.png);"
  />
  <img
    alt="Some icon"
    class="rn-bottom-1p0dtai rn-height-1pi2tsx rn-left-1d2f490 rn-opacity-1272l3b rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-width-13qz1uu rn-zIndex-1wyyakw"
    draggable="false"
    src="some-icon.png"
  />
</div>
`;
